ui-tests/src/main/java/io/jenkins/plugins/analysis/warnings/AnalysisSummary.java

```diff
@@ -0,0 +1,349 @@
+        }
+        return QualityGateResult.INACTIVE;
+    }
+
+    /**
+     * Gets the {@link WebElement} of the reset button.
+     *
+     * @return the button
+     * @throws org.openqa.selenium.NoSuchElementException
+     *         When there is no quality gate reset button.
+     */
+    public WebElement getQualityGateResetButton() throws org.openqa.selenium.NoSuchElementException {
+        for (WebElement result : results) {
+            if (result.getText().contains("Quality gate")) {
+                return result.findElement(by.id(id + "-resetReference"));
+            }
+        }
+        throw new org.openqa.selenium.NoSuchElementException("Quality gate reset button not found");
+    }
+
+    /**
+     * Checks if the quality gate reset button is present.
+     *
+     * @return True, if quality gate reset button is present.
+     */
+    public boolean hasQualityGateResetButton() {
+        try {
+            if (getQualityGateResetButton() != null) {
+                return true;
+            }
+        }
+        catch (org.openqa.selenium.NoSuchElementException ignored) {
+            // ignore and continue
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns a clickable WebElement (a-tag), by a part of the elements text.
+     *
+     * @param namePart
+     *         part of the visible text (should be unique within the item list)
+     *
+     * @return WebElement that belongs to the name part
+     */
+    private Optional<WebElement> findClickableResultEntryByNamePart(final String namePart) {
+        for (WebElement el : results) {
+            if (el.getText().contains(namePart)) {
+                return Optional.of(el.findElement(by.tagName("a")));
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Returns the complete visible text by a part of the elements text.
+     *
+     * @param namePart
+     *         part of the visible text (should be unique within the item list)
+     *
+     * @return String that belongs to the name part
+     */
+    public String findResultEntryTextByNamePart(final String namePart) {
+        for (WebElement el : results) {
+            if (el.getText().contains(namePart)) {
+                return el.getText();
+            }
+        }
+        return null;
+    }
+
+    private WebElement getSummary() {
+        return summary;
+    }
+
+    private WebElement getTitleResultLink() {
+        return summary.findElement(by.href(id));
+    }
+
+    private WebElement getTitleResultInfoLink() {
+        return summary.findElement(by.href(id + "/info"));
+    }
+
+    private WebElement getTitle() {
+        return title;
+    }
+
+    /**
+     * Determines which icon is shown to represent the info messages view.
+     */
+    public enum InfoType {
+        /** Info messages only. */
+        INFO("info-circle"),
+        /** Info and error messages. */
+        ERROR("exclamation-triangle");
+
+        private String iconName;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'iconName' could be replaced by a local variable.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)